### PR TITLE
Add node trace source map handling in dev mode

### LINKED ISSUE #2285: Show stacktraces with proper line numbers in src rather dist (SSR)
Currently when there's an error, it shows the line numbers of the `dist` version of the file.
We need to use sourcemaps or something other way to clearly show the correct file and the line number.

### PR DESCRIPTION
**Pull request by kpdecker**
Fixes #2285

### CODE REVIEW

**Comment by erikras** [comment]
This will literally save hours of dev time, as it is currently so non-obvious what line is causing an error. 👍 👍

**Comment by timneutkens** [comment]
This looks really good, let me give it a few spins before merging 👌

**Comment by ptomasroos** [comment]
This is brilliant @kpdecker !

File: server/index.js [hunk 1]
```diff
@@ -29,6 +29,14 @@ const blockedPages = {
 
 export default class Server {
   constructor ({ dir = '.', dev = false, staticMarkup = false, quiet = false, conf = null } = {}) {
+    // When in dev mode, remap the inline source maps that we genenrate within the webpack portion
```
Comment (timneutkens) [inline server/index.js]: small typo: genenrate

Comment (kpdecker) [reply server/index.js]: Fixed

Comment (timneutkens) [reply server/index.js]: ❤️

**Comment by arunoda** [comment]
This looks like something really great.
I'll give this a run and merge this.

**Comment by timneutkens** [comment]
Tried it last night, works 👌

**Comment by arunoda** [comment]
This is fantastic.
Thanks for the hard work @kpdecker

**Comment by arunoda** [comment]
I just published [3.2.0](https://github.com/zeit/next.js/releases/tag/3.2.0) which has this PR.

**Comment by dnakov** [comment]
This broke browser source maps.
To reproduce, run an example (e.g. with-absolute-imports) with next@3.2.0.
Source maps look like this:
```
null


// WEBPACK FOOTER //
// ./components/header.js
```

**Comment by kpdecker** [comment]
I can't really make sense of that output. What command is it from? What does your nextconfig look like?

**Comment by kpdecker** [comment]
And what browser?

**Comment by dnakov** [comment]
Chrome Version 60.0.3112.101 (Official Build) (64-bit) / MacOS 10.12.6 / Node 8.1.2

As I said above, just running the `with-absolute-imports` example in the repo with `next@3.2.0` and looking at the source maps in chrome devtools.

**Comment by timneutkens** [comment]
I'm thinking this is most likely related to the absolute imports example 🤔 cc @sergiodxa

**Comment by dnakov** [comment]
I just tried a random example -- with-redux, same result.
Downgraded to next@3.0.6 and the source maps came back.

**Comment by KClough** [comment]
I'm also experiencing the same issue as @dnakov , I reverted back to next@3.1.0 and no longer see an issue with sourcemaps while debugging.

**Comment by kpdecker** [comment]
Fixed in #2900.

**Pull request merged**